azure/leftovers.go

```diff
@@ -110,23 +108,34 @@ func NewLeftovers(logger logger, clientId, clientSecret, subscriptionId, tenantI
 		return Leftovers{}, errors.New("Missing tenant id.")
 	}
 
-	oauthConfig, err := adal.NewOAuthConfig(azurelib.PublicCloud.ActiveDirectoryEndpoint, tenantId)
-	if err != nil {
-		return Leftovers{}, fmt.Errorf("Creating oauth config: %s\n", err)
-	}
+	// oauthConfig, err := adal.NewOAuthConfig(azurelib.PublicCloud.ActiveDirectoryEndpoint, tenantId)
+	// if err != nil {
+	// 	return Leftovers{}, fmt.Errorf("Creating oauth config: %s\n", err)
+	// }
 
-	token, err := adal.NewServicePrincipalToken(*oauthConfig, clientId, clientSecret, azurelib.PublicCloud.ResourceManagerEndpoint)
+	// token, err := adal.NewServicePrincipalToken(*oauthConfig, clientId, clientSecret, azurelib.PublicCloud.ResourceManagerEndpoint)
+	// if err != nil {
+	// 	return Leftovers{}, fmt.Errorf("Creating service principal token: %s\n", err)
+	// }
+
+	credential, err := azidentity.NewClientSecretCredential(tenantId, clientId, clientSecret, nil)
 	if err != nil {
 		return Leftovers{}, fmt.Errorf("Creating service principal token: %s\n", err)
 	}
 
-	rg := resources.NewGroupsClient(subscriptionId)
-	rg.Authorizer = autorest.NewBearerAuthorizer(token)
+	rg, err := armresources.NewResourceGroupsClient(subscriptionId, credential, nil)
```

Comment:
Their [migration guide](https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/MIGRATION_GUIDE.md#authentication) recommends the client factory be created and then the clients pulled off the factory:
```
credential, err := azidentity.NewClientSecretCredential("<TenantId>", "<ClientId>", "<ClientSecret>", nil)
clientFactory, err := armresources.NewClientFactory(<subscription ID>, credential, &options)
if err != nil {
    log.Fatal(err)
}
client := clientFactory.NewResourceGroupsClient()
```

Reply:
the reason i did it this way is that i otherwise nott create multiple clientfactories.
one for `armresources` and one for `armnetwork`
if your more comfertable with that im happy to do so. but i though this way was a bit cleaner